### ISSUE TITLE
Live updating doesn't work

### ISSUE THREAD

**Issue by seagle0128** [emacsorphanage/emacs-git-gutter#146]
macOS 10.12.4
Emacs 25.2
GitGutter latest on melpa

```elisp
(custom-set-variables
 '(git-gutter:update-interval 2))
(global-git-gutter-mode 1)
```
But it doesn't work! Only change after saving the files.

Please check or let me know how to make it work kindly. Thanks!

**Comment by arichiardi** [comment]
I have noticed this too, cannot reproduce it though

**Comment by seagle0128** [comment]
Also reproduce with `emacs -Q`.

**Comment by ramnes** [comment]
I'm also noticing this. @syohex any hint on what to look at?

**Comment by seagle0128** [comment]
Any update for this issue? Is this project still alive?

**Comment by trev-dev** [comment]
"Error (use-package): git-gutter/:config: Symbol’s function definition is void: git-gutter:update-interval"

**Comment by seagle0128** [comment]
@trev-dev `git-gutter:update-interval` is a customization, not a function.

**Comment by trev-dev** [comment]
*mutters* Of course it is. I thought I had followed the config directions correctly. Let me try again, thanks!

**Comment by rossabaker** [comment]
It works for me in files of the base directory of the git repo, but not in subdirectories.  `git-gutter:live-update` is passing the base file name to `write-original-content`, which fails in `git-gutter:original-file-content`.

**Comment by leuven65** [comment]
It seems that the pull request is still not merged.

If you want to quickly fix this issue, maybe you can add following advice function to init file.
Here the function "projectile-project-root" is used to get the repo root, but it is also OK to use "vc-git-root".
` 
 (define-advice git-gutter:write-original-content (:around (orig-fun tmpfile filename &rest args) my-cmd)
    (let* ((prj-root (projectile-project-root filename))
           (rel-file (file-relative-name filename prj-root)))
      ;; (message "%s" rel-file)
      (apply orig-fun tmpfile rel-file args)
      ))
  ;; (advice-remove 'git-gutter:write-original-content #'git-gutter:write-original-content@my-cmd)
`

**Comment by seagle0128** [comment]
The project seems dead. I've switched to https://github.com/dgutov/diff-hl.
